runtime/autoload/provider.vim

```diff
@@ -19,3 +33,15 @@ function! provider#Poll(argv, orig_name, log_env) abort
   endtry
   throw remote#host#LoadErrorForHost(a:orig_name, a:log_env)
 endfunction
+
+" Attempts to append the lines in 'data' to file at 'log_path'
+"
+" returns true if the file was written to successfully
+function! s:WriteLines(orig_name, log_path, job, data, event)
```

Comment:
I'm actually somewhat concerned about how we're turning on capturing of stderr and never garbage collecting it. Once startup has been successful, the only thing the stderr stream is doing is leaking memory.

I'll try to rework the code a little so that we get the same message on crash but don't leak memory afterward.

Reply:
Hosts are not supposed to write to stderr after startup is successful, rather there is methods for the host to send messages over the API, so there should be any memory leak going on continuously, only a message getting saved and immediately discarded on a non-startup crash ("garbage collection" already happens by then). Rather than disabling stderr collection after startup (which could be done just by adding `chanclose(job, 'stderr')` ), it might be useful to show the stderr also at a later crash (or at least stash it in `:messages` with no immediate display if it is too jarring)